internal/dcache/cluster_manager/cluster_manager.go

```diff
@@ -1124,10 +1186,73 @@ func (cmi *ClusterManager) updateMVList(rvMap map[string]dcache.RawVolume, exist
 				break
 			}
 		}
+
+		// Call join mv and check if all rv's are able to join the mv with rv list
+		deleteRv, err := cmi.joinMV(mvName, existingMVMap[mvName])
+		if err != nil {
+			// If RPC server has not started we delete the new mv added and remove the node from the nodeToRvs map
+			// and redo distribution
+			// TODO :: Would prefer if we could validate this behaviour before allocating RVs to the Mv
+			if strings.Contains(err.Error(), "RPC server not started") {
+				log.Err("ClusterManagerImpl::updateMVList: RPC server not started, cannot join MV %s with RVs %v", mvName, existingMVMap[mvName])
+				delete(existingMVMap, mvName)
+				delete(nodeToRvs, rvMap[deleteRv].NodeId)
+				continue
+			} else {
+				log.Err("ClusterManagerImpl::updateMVList: Error joining MV %s with RV %s: %v", mvName, deleteRv, err)
+				return err
+			}
+		}
 	}
 
 	log.Debug("ClusterManager::updateMVList: existing MV map after phase#2: %v", existingMVMap)
 	// TODO :: arrange the map entries in lexicographical order
+	return nil
+}
+
+func (cmi *ClusterManager) joinMV(mvName string, rvs dcache.MirroredVolume) (string, error) {
```

Comment:
why do you call dcache.MirroredVolume type as 'rvs'?
JoinMV() should take just dcache.MirroredVolume as parameter. This described the MV completely including the component RVs, and that's what we need to make a JoinMV RPC call.